crates/red_knot/src/args.rs

```diff
@@ -299,3 +306,77 @@ pub(crate) enum TerminalColor {
     /// Never display colors.
     Never,
 }
+/// Either a path to a TOML configuration file (`knot.toml`),
+/// or a TOML `<KEY> = <VALUE>` pair
+/// (such as you might find in a `knot.toml` configuration file)
+/// overriding a specific configuration option.
+/// Later values override earlier ones, and values passed via this `--config` argument
+/// take precedence over values in other configuration files.
+#[derive(Debug, Clone)]
+pub(crate) struct ConfigsArg(Option<Options>);
+impl clap::FromArgMatches for ConfigsArg {
+    fn from_arg_matches(matches: &ArgMatches) -> Result<Self, Error> {
+        let cwd = {
+            let cwd = std::env::current_dir()?;
+            SystemPathBuf::from_path_buf(cwd).map_err(|path| Error::raw(ErrorKind::Io, format!("The current working directory `{}` contains non-Unicode characters. Red Knot only supports Unicode paths.", path.display())))?
+        };
+        let system = OsSystem::new(cwd);
+        let combined = matches
```

Comment:
Hmm. It's a bit awkward that we have to create a `system` out of the blue but I see why it's necessary. The only alternative would be is if we expose it using a thread local. 

I think this is fine for now

Reply:
@MichaReiser had a similar thought on instantiating `system` a second time there - agree it's awkward. one thought that crossed my mind was to separate out parsing the passed arguments from actually reading/parsing the config files so that could happen a bit further downstream and we could use the existing `system` instance. leaning towards leaving it like you say but if you have any strong feelings the other way happy to refactor

Reply:
There's a good chance this gets solved if we introduce the --config-file option because Args could then simply store a string and the loading and processing can happen later